File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/Menu.ts
import {MenuType} from "./MenuType";

export default interface Menu {

    type: MenuType
    icon: string | undefined
    caption: string
    journeyTypeId: string | undefined
    useJourneyTypeForId: boolean | undefined
    submenus: Menu[] | undefined
    visible: boolean | undefined
    remoteBaseUrl: string | undefined
    remoteUiId: string | undefined
    remoteMenuId: string | undefined

}

File: backend/webflux/dtos/src/main/java/io/mateu/dtos/Menu.java
package io.mateu.dtos;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.Collections;
import java.util.List;

public record Menu(
    MenuType type,
    String icon,
    String caption,
    String journeyTypeId,
    List<Menu> submenus,
    @JsonIgnore int order,
    boolean visible,
    String remoteBaseUrl,
    String remoteUiId,
    String remoteMenuId) {

  public Menu {
    submenus = Collections.unmodifiableList(submenus);
  }

  @Override
  public List<Menu> submenus() {
    return Collections.unmodifiableList(submenus);
  }
}
